core/services/pipeline/orm.go

```diff
@@ -58,99 +55,126 @@ func (o *orm) CreateSpec(ctx context.Context, tx *gorm.DB, pipeline Pipeline, ma
 	return spec.ID, errors.WithStack(err)
 }
 
-func (o *orm) CreateRun(db *gorm.DB, run *Run) (err error) {
+func (o *orm) CreateRun(db postgres.Queryer, run *Run) (err error) {
 	if run.CreatedAt.IsZero() {
 		return errors.New("run.CreatedAt must be set")
 	}
-	if err = db.Create(run).Error; err != nil {
-		return errors.Wrap(err, "error inserting pipeline_run")
-	}
+
+	ctx, cancel := postgres.DefaultQueryCtx()
+	defer cancel()
+	err = postgres.SqlxTransaction(ctx, db, func(tx *sqlx.Tx) error {
+		sql := `INSERT INTO pipeline_runs (pipeline_spec_id, meta, inputs, created_at, state)
+		VALUES (:pipeline_spec_id, :meta, :inputs, :created_at, :state)
+		RETURNING id`
+
+		query, args, e := tx.BindNamed(sql, run)
+		if e != nil {
```

Comment:
should be `err` here and in the prev line, no?

Reply:
ditto here